src/components/transactions/ExecuteTxButton/index.tsx

```diff
@@ -26,8 +26,10 @@ const ExecuteTxButton = ({
   const { setSelectedTxId } = useContext(ReplaceTxHoverContext)
   const safeSDK = useSafeSDK()
 
+  const isExpiredSwap = isSwapTxInfo(txSummary.txInfo) && txSummary.txInfo.status === 'expired'
```

Comment:
Maybe deserves a helper since you are doing the same in the TxDetails?

Reply:
Good idea, I will extract them into helper methods.

Reply:
done